File: Web/src/types/segment.ts
export type ISegmentItem = {
  id: number;
  trackId: number;
  name: string;
  duration: string;
  audioUrl: string;
  transcript: string;
  completed?: boolean;
};

export type ISegmentResponseItem = {
  id: number;
  name?: string;
  transcript: string;
  segmentDuration: string;
  orderInTrack: number;
  completed?: boolean;
};

export type IPostSegmentItem = {
  id: number;
  order?: number;
  startSec?: number;
  endSec?: number;
  transcript: string;
  audioUrl?: string;
  isCreate?: boolean;
};

export type IPostUpdateSegmentItem = {
  id: number | null;
  orderInTrack: number;
  startSec?: number;
  endSec?: number;
  transcript: string;
  isCreate?: boolean;
};

File: Server/Core/Modules/BasicPracticeModule/Entities/Segment.cs
using Core.Shared.Entities;

namespace Core.Modules.BasicListening.Entities
{
    public class Segment : ChangeableEntity<int>
    {
        public string Transcript { get; set; }
        public string AudioUrl { get; set; }
        public int OrderInTrack { get; set; }
        public double StartSec { get; set; }
        public double EndSec { get; set; }
        public int TrackId { get; set; }

        // Rela
        public Track Track { get; set; }
    }
}
